src/messages/out_of_band.rs

```diff
@@ -8,20 +8,19 @@ impl Message {
     ///
     /// * `body` - bytes of JSON serialized message body
     /// * `attachments` - optional set of `AttachmentBuilder` to be attached
-    ///
     pub fn as_out_of_band_invitation(
         mut self,
         body: impl AsRef<[u8]>,
         attachments: Option<Vec<AttachmentBuilder>>,
     ) -> Self {
-        self.jwm_header.typ = MessageType::DidcommRaw;
+        self.jwm_header.typ = MessageType::DidCommRaw;
         self.didcomm_header.m_type =
-            serde_json::to_string(&MessageType::DidcommInvitation).unwrap();
+            serde_json::to_string(&MessageType::DidCommInvitation).unwrap();
         if let Some(attachments) = attachments {
             for attachment in attachments {
                 self.apeend_attachment(attachment);
             }
         }
-        self.set_body(body.as_ref())
+        self.body(&String::from_utf8(body.as_ref().to_vec()).unwrap())
```

Comment:
I think `self.body(str::from_utf8(body.as_ref()).unwrap())` is better. Also, perhaps we should rename the method to `set_body`

Reply:
Updated the assignment part.

About the setter names:

- renamed it to `body` to keep the chain-able functions aligned, examples from the others chain-able setters:
  - reply_to
  - from
  - to
  - m_type
  - typ
  - kid
  - timed
  - and just spotted another exception: `set_didcomm_header`
- `set_body` and `set_didcomm_header` would then be the only chain-able setters with the `set_` prefix, open to rename it but then I'd suggest to rename the other setters as well, which _may_ not look as nice as it does now but looks better if written consistently, see code example below
- personally not a huge fan of the chain-able style and this may be due to my inexperience with it, by I would _expect_ a function called `set_x` to return either nothing or `Result<(), ...>`
- so before updating it, I'd like to confirm what we should do here:
  - update only `body` to `set_body` so we have it and `set_didcomm_header` to be the only ones with the `set_prefix`
  - prefix all chain-able setters that are named exactly like their fields with `set_`
  - keep `body` and drop the `set_` prefix from `set_didcomm_header`
- for choice 2: should we open a separate PR for this or should we handle it inthis PR?
- personally I would be for option 3 but open for any suggestion here :)